core/src/main/java/org/elasticsearch/index/query/HasChildQueryParser.java

```diff
@@ -33,7 +33,12 @@
  */
 public class HasChildQueryParser implements QueryParser<HasChildQueryBuilder> {
 
-    private static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField QUERY_FIELD = new ParseField("query", "filter");
+    public static final ParseField TYPE_FIELD = new ParseField("child_type", "type");
```

Comment:
So the docs for HasChildQuery talk about "type" only. The HasChildQueryBuilderTests.testParseFromJSON test uses "child_type". Given that as you mention HasParentQuery uses parent_type I think it would be best to stick with child_type.

After looking at the examples in our query DSL docs fixing the examples in the docs sounds like an additional PR to me.